src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -758,15 +765,27 @@ static TextSpan getFullSpan(TextSpan previousWhiteSpaceSpan, SyntaxTrivia trivia
             return previousWhiteSpaceSpan.IsEmpty ? trivia.FullSpan : TextSpan.FromBounds(previousWhiteSpaceSpan.Start, trivia.FullSpan.End);
         }
 
-        static void reportErrorFor(SourceFile sourceFile, SyntaxTrivia trivia)
+        void reportErrorFor(SyntaxTrivia trivia)
         {
             if (trivia.ContainsDiagnostics && trivia.IsKind(SyntaxKind.IgnoredDirectiveTrivia))
             {
-                throw new GracefulException(CliCommandStrings.CannotConvertDirective, sourceFile.GetLocationString(trivia.Span));
+                string location = sourceFile.GetLocationString(trivia.Span);
+                if (errors != null)
+                {
+                    errors.Add(new SimpleDiagnostic
+                    {
+                        Location = sourceFile.GetFileLinePositionSpan(trivia.Span),
+                        Message = string.Format(CliCommandStrings.CannotConvertDirective, location),
```

Comment:
Which language is this being localized in? I wonder if we'll need to pass something along to give you a hint.